dali/dfu/dfuutil.cpp

```diff
@@ -868,10 +897,169 @@ class CDFUhelper: public CInterface, implements IDFUhelper
+                    sublname.set(subfilename.str());
+                    if (!sublname.isExternal()&&!sublname.isForeign())
+                    {
+                        StringBuffer subquery;
+                        sublname.makeFullnameQuery(subquery, DXB_File, true);
+                        Owned<IRemoteConnection> subconn = querySDS().connect(subquery.str(),myProcessSession(),RTM_LOCK_WRITE, FOREIGN_DALI_TIMEOUT);
+                        if (!subconn)
+                        {
+                            sublname.makeFullnameQuery(subquery.clear(), DXB_SuperFile, true);
+                            subconn.setown(querySDS().connect(subquery.str(),myProcessSession(),0, FOREIGN_DALI_TIMEOUT));
+                        }
+                        if (!subconn)
+                        {
+                            ERRLOG("SuperFile %s is missing sub-file file %s",superName.str(),subfilename.str());
+                            continue;
+                        }
+                        subnum++;
+
+                        Owned<IPropertyTree> subroot = subconn->getRoot();
+                        Owned<IPropertyTreeIterator> iter = subroot->getElements("SuperOwner");
+                        StringBuffer pname;
+                        bool parentok=false;
+                        ForEach(*iter)
+                        {
+                            iter->query().getProp("@name",pname.clear());
+                            if (strcmp(pname.str(),superName.str())==0)
+                                parentok = true;
+                            else
+                            {
+                                CDfsLogicalFileName sdlname;
+                                sdlname.set(pname.str());
+                                StringBuffer sdquery;
+                                sdlname.makeFullnameQuery(sdquery, DXB_SuperFile, true);
+                                Owned<IRemoteConnection> sdconn = querySDS().connect(sdquery.str(),myProcessSession(),0, FOREIGN_DALI_TIMEOUT);
+                                if (!conn)
+                                    WARNLOG("SubFile %s has missing owner superfile %s",sublname.get(),sdlname.get());
+                            }
+                        }
+                        if (!parentok)
+                        {
+                            WARNLOG("SubFile %s is missing link to Superfile %s",sublname.get(),superName.str());
+                            Owned<IPropertyTree> t = createPTree("SuperOwner");
+                            t->setProp("@name",superName.str());
+                            subroot->addPropTree("SuperOwner",t.getClear());
+
+                        }
+                        subconn->commit();
+                        subconn->close();
+                    }
+                }
+            }
+            PROGLOG("%d of %d sub file(s) reference to superfile %s are fixed.",subnum, numSubfiles, superName.str());
```

Comment:
There needs to be a similar approach to the handling of a standard file for super files.
IDFUhelper should not interact with Dali directly or deal with the inner workings of the file representation.
It use DFS methods to create objects (that validate if necessary).

There isn't a current [DFS] way to reattach a exported super file representation - but for this feature there should be.
The way I would do it, is to add a IDistributedFileDirectory::createNewSuper, which takes the IPropertyTree.
It can do some basic validation, then create a [unattached] CDistributedSuperFile / IDistributedSuperFile.
You can then use the existing methods to deal with SuperOwner etc., without going directly to the raw XML via IPT.

When validated, you can attach() it.